api/src/main/java/org/openmrs/api/impl/UserServiceImpl.java

```diff
@@ -751,21 +753,34 @@ public User getUserByActivationKey(String activationKey) {
 	}
 	
 	/**
+	 * @throws MessageException
 	 * @see org.openmrs.api.UserService#setUserActivationKey(org.openmrs.User)
-	 * 
 	 */
 	@Override
 	public User setUserActivationKey(User user) {
 		String token = RandomStringUtils.randomAlphanumeric(20);
 		long time = System.currentTimeMillis() + getValidTime();
 		String hashedKey = Security.encodeString(token);
-		String activationKey = hashedKey+":"+time;
+		String activationKey = hashedKey + ":" + time;
 		LoginCredential credentials = dao.getLoginCredential(user);
 		credentials.setActivationKey(activationKey);	
 		dao.setUserActivationKey(credentials);	
-			
-		//Send Email with unhashed  activation key and Date:
 		
+		String link = Context.getAdministrationService().getGlobalProperty(OpenmrsConstants.GP_HOST_URL)
+		        .replace("activationkey", "activationkey=" + token);
```

Comment:
Please try to use constants where necessary, the placeholder needs to look explicit, use something like  **{actvitationKey}**, and you should replace the placeholder with only the key, nothing else.